minorminer/utils/zephyr/node_edge.py

```diff
@@ -0,0 +1,635 @@
+                    )
+        self._ccoord = new_ccoord
+
+    @staticmethod
+    def get_coord(coord: tuple[int]) -> CartesianCoord | ZephyrCoord:
+        """Takes a tuple[int] and returns the corresponding CartesianCoord
+        or ZephyrCoord"""
+        if (not isinstance(coord, tuple)) or (not all(isinstance(c, int) for c in coord)):
+            raise TypeError(f"Expected {coord} to be a tuple[int], got {coord}")
+        if any(c < 0 for c in coord):
+            raise ValueError(f"Expected elements of coord to be non-negative, got {coord}")
+        len_coord = len(coord)
+        if len_coord in (2, 3):
+            x, y, *k = coord
+            if x % 2 == y % 2:
+                raise ValueError(f"Expected x, y to differ in parity, got {x, y}")
+            return CartesianCoord(x=x, y=y) if len(k) == 0 else CartesianCoord(x=x, y=y, k=k[0])
+        if len_coord in (4, 5):
+            u, w, *k, j, z = coord
+            for var, val in {"u": u, "j": j}.items():
+                if not val in [0, 1]:
+                    raise ValueError(f"Expected {var} to be in [0, 1], got {val}")
+            return (
+                ZephyrCoord(u=u, w=w, j=j, z=z)
+                if len(k) == 0
+                else ZephyrCoord(u=u, w=w, k=k[0], j=j, z=z)
+            )
+        raise ValueError(f"coord can have length 2, 3, 4 or 5, got {len_coord}")
+
+    @property
+    def zcoord(self) -> ZephyrCoord:
+        """Returns ZephyrCoordinate corresponding to ccoord"""
+        return cartesian_to_zephyr(self._ccoord)
+
+    def is_quo(self) -> bool:
+        """Decides if the ZNode object is quotient"""
+        return (self._ccoord.k is None) and (self._shape.t is None)
+
+    def to_quo(self) -> ZNode:
+        """Returns the quotient ZNode corresponding to self"""
+        qshape = ZShape(m=self._shape.m)
+        qccoord = CartesianCoord(x=self._ccoord, y=self._ccoord)
+        return ZNode(coord=qccoord, shape=qshape, convert_to_z=self.convert_to_z)
+
+    @property
+    def direction(self) -> int:
+        """Returns direction, 0 or 1"""
+        return self._ccoord.x % 2
+
+    def is_vertical(self) -> bool:
+        """Decides whether self is a vertical qubit"""
+        return self.direction == 0
+
+    def is_horizontal(self) -> bool:
+        """Decides whether self is a horizontal qubit"""
+        return self.direction == 1
+
+    def node_kind(self) -> str:
+        """Returns the node kind, vertical or horizontal"""
+        return "vertical" if self.is_vertical() else "horizontal"
+
+    def neighbor_kind(
+        self,
+        other: ZNode,
+    ) -> str | None:
+        """Returns the kind of coupler between self and other,
+        'internal', 'external', 'odd', or None."""
+        if not isinstance(other, ZNode):
+            other = ZNode(other)
+        if self._shape != other._shape:
+            return
```

Comment:
Conventions are to return `None` explicitly if function returns anything. See https://stackoverflow.com/a/15300671.
```suggestion
            return None
```
Same in other places in this method.

Reply:
Done in commit [fd78154](https://github.com/dwavesystems/minorminer/pull/261/commits/fd78154bb8a05b02fcd4f8cc75569591166c3272)